methods.py

```diff
@@ -1,33 +1,38 @@
-from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions as EC
 from selenium.webdriver.support.wait import WebDriverWait
 
 
 class Methods(object):
-
+    """Webdriver wrapper"""
     def __init__(self, driver):
         self.driver = driver
 
-    def get_element(self, locator_type, locator, numb_list=0):
-        # This function seeks for elements
-        types_By = {
-            'id': By.ID,
-            'css': By.CSS_SELECTOR}
-        element = self.driver.find_elements(types_By[locator_type], locator)
-        return element[numb_list]
-
-    def click_element(self, locator_type, locator, numb_list=0):
-        element = self.get_element(locator_type, locator, numb_list)
-        element.click()
-
-
-    def clear_element(self, locator_type, locator):
-        element = self.get_element(locator_type, locator)
-        element.clear()
-
-    def send_keys(self, locator_type, locator, keys):
-        element = self.get_element(locator_type, locator)
-        element.send_keys(keys)
-
-
-            
+    def get_elements(self, locator):
+        """Returns all elements for the specific locator"""
+        return self.driver.find_elements(*locator)
+
+    def click_element(self, locator, elem_number=0):
+        """Clicks on the element with number elem_number"""
+        WebDriverWait(self.driver, 10).until(EC.element_to_be_clickable(locator))
+        elements = self.get_elements(locator)
+        elements[elem_number].click()
+
+    def click_element_by_text(self, locator, text_value):
+        """Clicks on the element with text attribute text_value"""
+        WebDriverWait(self.driver, 10).until(EC.element_to_be_clickable(locator))
```

Comment:
Let's extract wait time into constant and place it in BasePage class.

Reply:
You mean TIMEOUT = 10 ?

Reply:
Yeap